endpoints/cookie_sync.go

```diff
@@ -285,17 +287,33 @@ func (c *cookieSyncEndpoint) writeParseRequestErrorMetrics(err error) {
 }
 
 func (c *cookieSyncEndpoint) setLimit(request cookieSyncRequest, cookieSyncConfig config.CookieSync) cookieSyncRequest {
-	if request.Limit <= 0 && cookieSyncConfig.DefaultLimit != nil {
-		request.Limit = *cookieSyncConfig.DefaultLimit
+	limit := getEffectiveLimit(request.Limit, cookieSyncConfig.DefaultLimit)
+	maxLimit := getEffectiveMaxLimit(cookieSyncConfig.MaxLimit)
+	if maxLimit < limit {
+		request.Limit = &maxLimit
+	} else {
+		request.Limit = &limit
 	}
-	if cookieSyncConfig.MaxLimit != nil && (request.Limit <= 0 || request.Limit > *cookieSyncConfig.MaxLimit) {
-		request.Limit = *cookieSyncConfig.MaxLimit
+	return request
+}
+
+func getEffectiveLimit(reqLimit *int, defaultLimit *int) int {
+	limit := reqLimit
+	if reqLimit == nil && defaultLimit != nil {
+		limit = defaultLimit
 	}
-	if request.Limit < 0 {
-		request.Limit = 0
+	if limit == nil || *limit <= 0 || *limit > math.MaxInt32 {
+		limit = ptrutil.ToPtr(math.MaxInt32)
 	}
+	return *limit
+}
```

Comment:
Since `limit` is of type `int` as opposed to `int32` or `int64`, can't it be greater than `math.MaxInt32` depending on the underlying architecture?
In that case maybe this should be `math.MaxInt`.

Reply:
That's a good point, yes - let's use `math.MaxInt`.